### PR TITLE
Hide visualization files

### LINKED ISSUE #934: hide visualization files not to be opened
It is crucial to open the .pvd and not the other files and finding them in paraview is painful.

Maybe we should write the .vtu/.pvtu files inside a subdirectory like output/solution/ ?

### PR DESCRIPTION
**Pull request by tjhei**
supersedes #1076 and fixes #934 .

### CODE REVIEW

**Comment by tjhei** [comment]
ready to review.

File: include/aspect/utilities.h [hunk 1]
```diff
@@ -220,6 +220,18 @@ namespace aspect
...
+     * Create directory @p pathname, optionally printing a message.
+     *
+     * @param pathname String that contains path to create. '/' is used as
+     * directory separator.
+     * @param comm MPI communicator, used to limit creation of directory to
+     * processor 0.
+     * @param silent Print a nicely formatted message on processor 0 if set
+     * to true.
+     */
+    void create_directory(std::string pathname, const MPI_Comm &comm, bool silent);
```
Comment (bangerth) [inline include/aspect/utilities.h]: make first argument a const reference. break arguments onto separate lines

File: source/particle/output/ascii.cc [hunk 2]
```diff
@@ -34,14 +35,29 @@ namespace aspect
       {}
 
       template <int dim>
+      void ASCIIOutput<dim>::initialize ()
+      {
+        std::string particle_subdirectory = this->get_output_directory() + "particles/";
```
Comment (bangerth) [inline source/particle/output/ascii.cc]: make const

Comment (bangerth) [inline source/particle/output/ascii.cc]: or just put it into the call on the next line

File: source/particle/output/hdf5.cc [hunk 3]
```diff
@@ -61,6 +62,16 @@ namespace aspect
       }
 
       template <int dim>
+      void HDF5Output<dim>::initialize ()
+      {
+        std::string particle_subdirectory = this->get_output_directory() + "particles/";
+
+        aspect::Utilities::create_directory (particle_subdirectory,
```
Comment (bangerth) [inline source/particle/output/hdf5.cc]: same here

File: source/particle/output/vtu.cc [hunk 4]
```diff
@@ -35,19 +36,30 @@ namespace aspect
       {}
 
       template <int dim>
+      void VTUOutput<dim>::initialize ()
+      {
+        particle_dirname = "particles/";
+        particle_subdirectory = this->get_output_directory() + particle_dirname;
```
Comment (bangerth) [inline source/particle/output/vtu.cc]: why make these member variables here when you had them as just local variables in the ASCIIData class? My personal preference would to not have them as member variables, but I'm mostly interested in consistency.

File: source/utilities.cc [hunk 5]
```diff
@@ -387,6 +388,47 @@ namespace aspect
       return 0;
     }
 
+    void create_directory(std::string pathname, const MPI_Comm &comm, bool silent)
```
Comment (bangerth) [inline source/utilities.cc]: make first argument a const reference. break arguments onto separate lines

File: source/particle/output/vtu.cc (hunk 4 above)
Comment (tjhei) [inline source/particle/output/vtu.cc]: sure, I missed those.

**Pull request merged**